documentation/Feedback.md

```diff
@@ -0,0 +1,267 @@
+# Documentation: ContraBot Feedback Feature
+
+## Overiew 
+
+After a user completes the political test, the bot will, after one week, check if the user would be willing to provide feedback. This feedback is then saved to a Google Sheet for further analysis.
+
+### Reset User Context in Database
+
+Once the political survey is completed, the bot resets the user's context in the database, setting the current question index back to the beginning and noting the completion time.
+
+**test-command.ts**
+```typescript
+await db.db('contrabot').collection("users").updateOne(
+    { userId: interaction.user.id },
+    {
+        $set: {
+            currentQuestionIndex: 0,  // Reset to first question
+            completionTime: DateTime.now().toISO(), // Set completion time
+        }
+    }
+);
+```
+
+### Feedback Request Function
+
+The function `checkForFeedbackRequests` is executed periodically to check which users finished their survey a week ago and have not been prompted for feedback.
+
+1. It first calculates the date a week ago.
+2. It then fetches users from the database who finished the survey more than a week ago and haven't been asked for feedback.
+3. It sends a feedback request message to those users and updates the database to ensure they won't be prompted again.
+4. It creates a button to start the survey and sends it to the user.
+
+**test-command.ts**
+```typescript
+const checkForFeedbackRequests = async () => {
+    const now = DateTime.now();
+    const oneWeekAgo = now.minus({ days: 7 });
+
+    const users = await db.db('contrabot').collection("users").find({
+        completionTime: { 
+            $lt: oneWeekAgo.toISO()
+        },
+        feedbackRequestSent: { $ne: true } // This ensures that you don't ask for feedback multiple times
+    }).toArray();
+
+    // Create a button to start the survey
+    const startSurveyButton = new ButtonBuilder()
+        .setCustomId('start_survey')
+        .setLabel('Start Survey')
+        .setStyle(ButtonStyle.Primary);
+
+    const actionRow = new ActionRowBuilder<ButtonBuilder>()
+        .addComponents(startSurveyButton);
+
+    for (const user of users) {
+        const discordUser = await client.users.fetch(user.userId);
+        if (discordUser) {
+            await discordUser.send({
+                content: `
+                Hallo 👋, vor einer Woche hast du den Test ausgefüllt. 
+                Wir können Contraversum nur verbessern durch Feedback von unseren Nutzern. 
+                Daher wäre es ein wichtiger Beitrag für das Projekt und damit auch für die depolarisierung 
```

Comment:
'depolarisierung' needs to be capitalized, maybe 'politische Depolarisierung' would be a bit better.